src/flightctl/context.rs

```diff
@@ -1,141 +1,208 @@
 use super::aws;
 use super::config::{Auth, AuthConfig, Cluster, ClusterConfig, Config, Context, Release};
-use super::kubeconfig;
-use base64;
-use std::io::Write;
-use tempfile::NamedTempFile;
+use super::kubeconfig_writer;
+use kube::config::{
+    AuthInfo, ExecConfig, Kubeconfig, KubeconfigError, NamedAuthInfo, NamedCluster,
+};
+use std::collections::HashMap;
 
 pub fn prepare(config: &Config, release: &Release) -> anyhow::Result<()> {
     let context = config.find_context(&release)?;
-    log::debug!("Looking for context {}", context.name);
-    let context_exists = kubeconfig::context_exists(&context)?;
+    let auth = config.find_auth(context)?;
+    let cluster = config.find_cluster(context)?;
+    let kubeauth = build_auth(context, auth, cluster);
+    let kubecluster = build_cluster(cluster, auth)?;
 
-    if context_exists {
-        log::info!("Using existing context: {}", context.name);
+    let kubeconfig = read_kubeconfig()?;
+    log::debug!(
+        "Loaded Kubernetes configuration successfully: {:?}",
+        kubeconfig
+    );
+    ensure_auth(&kubeconfig, &kubeauth)?;
+    ensure_cluster(&kubeconfig, &kubecluster)?;
+    ensure_context(&kubeconfig, &context)
```

Comment:
Would it make sense to try to make these more transactional, e.g. if `ensure_cluster` fails, we'd rollback `ensure_auth`?

Reply:
I think it would actually be more convenient for the auth/cluster to be left in the config for debugging. Otherwise you can't rerun the failed command yourself to see what's happening.